### ISSUE TITLE
Ignore ParticleSystem without materials

### ISSUE THREAD

**Issue by sp-alberto-uriarte** [mob-sakai/ParticleEffectForUGUI#118]
Sometimes our VFX artists uses an "empty" particle system as a root to see all nested particle system running at once.
It would be great if particle systems without materials are ignored. To do that I applied the following changes 

```
Index: Project/Packages/ParticleEffectForUGUI/Scripts/UIParticle.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Project/Packages/ParticleEffectForUGUI/Scripts/UIParticle.cs	(revision e21724ee1ad550bb985c12433b00d48e6e92f77a)
+++ Project/Packages/ParticleEffectForUGUI/Scripts/UIParticle.cs	(date 1605787646189)
@@ -209,6 +209,8 @@
             if (!root) return;
             root.GetComponentsInChildren(particles);
 
+            particles.RemoveAll(HasNoMaterial);
+
             foreach (var ps in particles)
             {
                 var tsa = ps.textureSheetAnimation;
@@ -222,6 +224,11 @@
             SetMaterialDirty();
         }
 
+        static bool HasNoMaterial(ParticleSystem particleSystem)
+        {
+            return particleSystem.GetComponent<ParticleSystemRenderer>()?.sharedMaterial == null;
+        }
+
         protected override void UpdateMaterial()
         {
             // Clear mask materials.
```

**Comment by mob-sakai** [comment]
Thank you for the feature request!

**Comment by mob-sakai** [comment]
Change as follows.
* Ignore rendering of particle systems that do not have a SharedMaterial and TrailMaterial.

**Comment by mob-sakai** [comment]
Fixed in `v3.3.0`
